prqlc/prql-compiler/src/semantic/resolver/types.rs

```diff
@@ -500,10 +462,65 @@ pub(crate) fn normalize_type(ty: Ty) -> Ty {
             Ty { kind, ..ty }
         }
 
+        TyKind::Array(items_ty) => Ty {
+            kind: TyKind::Array(Box::new(normalize_type(*items_ty))),
+            ..ty
+        },
+
         kind => Ty { kind, ..ty },
     }
 }
 
+/// Sinks union into arrays and tuples.
+/// [A] || [B] -> [A || B]
```

Comment:
Nitty but should we either use `|` or `||` throughout?

Reply:
It should be `||`, we've decided on this already.

Reply:
Totally fine with me, FYI we do have lots of `|`: https://github.com/PRQL/prql/pull/4013/files#diff-5f3418da7152d1109aefd6a955a037dee9b32049a29f5f750ad52160fc766d40R247